commown_devices/models/project_task.py

```diff
@@ -1,4 +1,7 @@
 from odoo import _, api, fields, models
+from odoo.exceptions import Warning
+
+CHECK_CONTRAT_QUANT_NB_STAGE_IDS = [223, 226]
```

Comment:
I think we should use XML id here.

Reply:
ok but on those elements the get_xml_id() function return empty :
In [14]: env["project.task.type"].browse(226).get_xml_id()
Out[14]: {226: ''}

Should I use the id returned by " .export_data(['id'])" ?

Reply:
You must add the project.task.type entities you need in a data file in the module, use the xml ids in the code (never use hard-coded constants in a module). If you insert the xml ids in the production database before updating the module, the entities will not be recreated.

Reply:
Moreover there is no unittest in your code for the added "feature". This would have rung a bell: hard-coded ids in the test database are certainly not the same as in the production DB.

Reply:
> You must add the project.task.type entities you need in a data file in the module, use the xml ids in the code (never use hard-coded constants in a module). If you insert the xml ids in the production database before updating the module, the entities will not be recreated.

Not only.
well two things : 
- What you said, create a noupdate data in a xml file, loaded in the `data` key of the __manifest__.py -> make the CI working. (and so, it's possible to add test, but also constraint to prevent deletion, etc...)
- Create a pre-migration script in a migration folder to create manually your ir.model.data items to point on [223, 226] + bump module version -> make your production working. (if not, you will create duplicates, because I guess `223, 226` still exists on your production.)

Reply:
yes, this is a more precise way to explain what my second sentence says, although I do not use pre-migration scripts for this kind of purpose anymore, but external throw-away scripts or even manual xml id creation using the UI, to avoid polluting the module's code and git history with ephemeral code, but this is another story.